samcli/commands/local/cli_common/invoke_context.py

```diff
@@ -345,6 +349,25 @@ def _clean_running_containers_and_related_resources(self) -> None:
         cast(WarmLambdaRuntime, self.lambda_runtime).clean_running_containers_and_related_resources()
         cast(RefreshableSamFunctionProvider, self._function_provider).stop_observer()
 
+    def _add_account_id_to_global(self) -> None:
+        """
+        Attempts to get the Account ID from the current session
+        If there is no current session, the standard parameter override for
+        AWS::AccountId is used
+        """
+        client_provider = get_boto_client_provider_with_config(region=self._aws_region, profile=self._aws_profile)
+
+        sts = client_provider("sts")
+
+        try:
+            account_id = sts.get_caller_identity().get("Account")
+            if account_id:
+                if self._global_parameter_overrides is None:
+                    self._global_parameter_overrides = {}
+                self._global_parameter_overrides["AWS::AccountId"] = account_id
+        except (NoCredentialsError, TokenRetrievalError):
+            LOG.warning("No current session found, using default AWS::AccountId")
+
```

Comment:
If this function is ok to error, can you also add a generic exception handling right now? If theres no exception handling here, if an error occurs outside of the 2 defined errors, there is potential for this function to panic the execution

Reply:
I caatch these two exceptions because they can be the consequences of
- Not having a `~/.aws/credentials` or `~/.aws/config` file
- Valid configuration but token is expired
If anything else causes an error, we don't know what's caused it and in my opinion the execution should panic because we don't know how to handle it.
Thoughts? I mean, I'm happy to put in a generic `except` and continue but I'm not sure if that's what we should do here

Reply:
We might want to consider catching the `ClientError` exception here too. Using expired credentials, botocore returns:

```
botocore.exceptions.ClientError: An error occurred (ExpiredToken) when calling the GetCallerIdentity operation: The security token included in the request is expired
```

You had mentioned that other tests start failing once the `ClientError` was caught, maybe we could look at fixing those?

Reply:
What I've done is add catching `ClientError` to the new method, and patching the new method for the existing tests so it's just a stub; they don't need a live connection so inserting a live account ID is irrelevant. All tests are passing now

Reply:
I also put back in some definitions of `aws_profile` I had taken out of some of those tests. Now the only differences to upstream are the patching of `_add_account_id_to_global` and the extra tests for the new code.

Reply:
@lucashuy What I've done is add catching `ClientError` to the new method, and patching the new method for the existing tests so it's just a stub; they don't need a live connection so inserting a live account ID is irrelevant. All tests are passing now
I also put back in some definitions of `aws_profile` I had taken out of some of those tests. Now the only differences to upstream are the patching of `_add_account_id_to_global` and the extra tests for the new code.